magda-indexer/src/main/scala/au/csiro/data61/magda/indexer/external/registry/WebhookApi.scala

```diff
@@ -32,23 +34,38 @@ class WebhookApi(indexer: SearchIndexer)(implicit system: ActorSystem, config: C
     magdaRoute {
       post {
         entity(as[WebHookPayload]) { payload =>
-          payload.records match {
-            // TODO: Handle registry config not found
-            case Some(records) =>
-              val dataSets = records.map(record => try {
+          val events = payload.events match {
+            case None       => List()
+            case Some(list) => list
+          }
+
+          val idsToDelete = events.filter(_.eventType == EventType.DeleteRecord)
```

Comment:
What if the record is deleted and quickly recreated (i.e. by events in the same webhook payload)?

Reply:
Good point. Wrote a test for this and somehow it works but I think that's just luck - have made the execution order explicit (delete everything then create everything).